pkg/errors/errors.go

```diff
@@ -11,13 +11,13 @@ type ErrorWithStatusCode struct {
 	statusCode int
 }
 
-func NewErrorWithStatusCode(err error, statusCode int) *ErrorWithStatusCode {
-	return &ErrorWithStatusCode{
+func NewErrorWithStatusCode(err error, statusCode int) ErrorWithStatusCode {
+	return ErrorWithStatusCode{
 		error:      err,
 		statusCode: statusCode,
 	}
 }
 
-func (e *ErrorWithStatusCode) StatusCode() int {
+func (e ErrorWithStatusCode) StatusCode() int {
```

Comment:
Why not a pointer receiver? This'll copy the entire structure each time it's called

Reply:
Use ErrorWithStatusCode by value to avoid breaking compatibility in a really bad way.

Reply:
Also, the structure contains just an int and an interface, so very little data.